packages/map-template/src/components/Directions/Directions.jsx

```diff
@@ -255,78 +246,55 @@ function Directions({ isOpen, onBack, onSetSize, snapPointSwiped }) {
-                                    </div>}
-                                <div className="directions__content">
-                                    <div className='directions__name'>
-                                        {directions?.destinationLocation.properties.name}
-                                    </div>
-                                    <mi-location-info level={t('Level')} ref={destinationInfoElement} show-external-id={false} />
-                                </div>
-                            </div>
-                        }
-                    </div>
-                </div>
-            </div>
-            <div ref={guideElement} className="directions__guide">
-                <div className="directions__route">
-                    <div className="directions__metrics">
-                        <div className="directions__distance">
-                            {travelMode === travelModes.WALKING && <WalkingIcon />}
-                            {travelMode === travelModes.DRIVING && <DriveIcon />}
-                            {travelMode === travelModes.BICYCLING && <BikeIcon />}
-                            <div>{t('Distance')}:</div>
-                            <div className="directions__meters">{totalDistance && <mi-distance unit={distanceUnitSystem} meters={totalDistance} />}</div>
-                        </div>
-                        <div className="directions__time">
-                            <ClockIcon />
-                            <div>{t('Estimated time')}:</div>
-                            <div className="directions__minutes">{totalTime && <mi-time translations={JSON.stringify({ days: t('d'), hours: t('h'), minutes: t('min') })} seconds={totalTime} />}</div>
-                        </div>
-                    </div>
-                    {kioskLocation && supportsUrlParameters && <button className='directions__qr-code' onClick={() => setShowQRCodeDialog(true)}><QRCode /> {t('Scan QR code')}</button>}
-                </div>
-                <hr></hr>
+        <div className="directions" style={{ display: isKioskContext ? 'grid' : 'block' }}>
+            <div className="directions__steps">
+                <div className="directions__minutes">{totalTime && <mi-time translations={JSON.stringify({ days: t('d'), hours: t('h'), minutes: t('min') })} seconds={totalTime} />}</div>
                 <RouteInstructions
                     steps={getRouteSteps()}
                     originLocation={directions?.originLocation}
                     onNextStep={() => onNext()}
                     isOpen={isOpen}
                     onPreviousStep={() => onPrevious()}
-                    onFitCurrentDirections={() => onFitCurrentDirections()}
-                >
+                    onFitCurrentDirections={() => onFitCurrentDirections()} >
                 </RouteInstructions>
             </div>
+            {kioskLocation && isDesktop &&
```

Comment:
Could we use the new custom hook here?

Reply:
Yes, I think so, replaced it with the custom hook. 😄